BackofficeBundle/Resources/public/ecmascript/Lib/jquery.initialize.js

```diff
@@ -1,45 +1,68 @@
 /*!
- * jQuery initialize - v1.0.0 - 12/14/2016
+ * Open Orchestra jQuery initialize
+ *
+ * Refacto of jQuery initialize - v1.0.0 - 12/14/2016
  * https://github.com/adampietrasiak/jquery.initialize
  *
  * Copyright (c) 2015-2016 Adam Pietrasiak
  * Released under the MIT license
  * https://github.com/timpler/jquery.initialize/blob/master/LICENSE
  */
 ;(function ($) {
-    // MutationSelectorObserver represents a selector and it's associated initialization callback.
-    var MutationSelectorObserver = function (selector, callback) {
-        this.selector = selector;
-        this.callback = callback;
-    };
+    var msobservers = {
+        'callbacks': {},
 
-    // List of MutationSelectorObservers.
-    var msobservers = [];
-    msobservers.initialize = function (selector, callback) {
-
-        // Wrap the callback so that we can ensure that it is only
-        // called once per element.
-        var seen = [];
-        callbackOnce = function () {
-            if (seen.indexOf(this) == -1) {
-                seen.push(this);
-                $(this).each(callback);
-            }
-        };
+        'initialize': function (selector, callback) {
+            // Wrap the callback so that we can ensure that it is only called once per element.
+            var seen = [];
+            callbackOnce = function () {
+                if (seen.indexOf(this) == -1) {
+                    seen.push(this);
+                    $(this).each(callback);
+                }
+            };
 
-        // See if the selector matches any elements already on the page.
-        $(selector).each(callbackOnce);
+            // See if the selector matches any elements already on the page.
+            $(selector).each(callbackOnce);
+
+            // Then, add it to the list of selector observers.
+            this.addCallback(selector, callbackOnce);
+        },
+
+        'addCallback': function (selector, callback) {
+            if (typeof this.callbacks[selector] == 'undefined') {
+                this.callbacks[selector] = [];
+            }
+            this.callbacks[selector].push(callback);
+        },
 
-        // Then, add it to the list of selector observers.
-        this.push(new MutationSelectorObserver(selector, callbackOnce));
+        'removeSelector': function(selector) {
+            if (this.callbacks.hasOwnProperty(selector) !== 'undefined') {
+                delete(this.callbacks[selector]);
```

Comment:
"delete" will keep entries to "undefined". "splice" function will optimize the process.